File: 1542-find-longest-awesome-substring/1542-find-longest-awesome-substring.cpp
class Solution {
public:
     int longestAwesome(string s) {
        int res = 0, cur = 0, n = s.size();
        vector<int> seen(1024, n);
        seen[0] = -1;
        for (int i = 0; i < n; ++i) {
            cur ^= 1 << (s[i] - '0');
            for (int a = 0; a < 10; ++a)
                res = max(res, i - seen[cur ^ (1 << a)]);
            res = max(res, i - seen[cur]);
            seen[cur] = min(seen[cur], i);
        }
        return res;
    }
};

File: 1542-find-longest-awesome-substring/1542-find-longest-awesome-substring.java
class Solution {
    public int longestAwesome(String s) {
        int res = 0, cur = 0, n = s.length(), seen[] = new int[1024];
        Arrays.fill(seen, n);
        seen[0] = -1;
        for (int i = 0; i < n; ++i) {
            cur ^= 1 << (s.charAt(i) - '0');
            for (int a = 0; a < 10; ++a)
                res = Math.max(res, i - seen[cur ^ (1 << a)]);
            res = Math.max(res, i - seen[cur]);
            seen[cur] = Math.min(seen[cur], i);
        }
        return res;
    }
}